src/main/java/com/squareup/kotlinpoet/TypeVariableName.kt

```diff
@@ -66,10 +68,6 @@ class TypeVariableName private constructor(
       require(variance == null || variance.isOneOf(KModifier.IN, KModifier.OUT)) {
         "$variance is an invalid variance modifier, the only allowed values are in and out!"
       }
-      require(bounds.isNotEmpty()) {
```

Comment:
... and if you push it out, you can restore this check.

Reply:
I don't think so. The awkward part in moshi was that we were always passing in a bounds list, it just might sometimes be empty. In this case, that would still fail this check every time, whereas I thought the goal was to check for empty inputs and coerce them to a `listOf(NULLABLE_ANY)` 🤔